src/pages/ScreenRouting/mediaScreens/MediaSeries/MediaSeries.tsx

```diff
@@ -1,21 +1,302 @@
+          <script type="application/ld+json">{generateEpisodeJSONLD(series, seriesMedia, episode, episodeMetadata)}</script>
+        ) : null}
+      </Helmet>
+      <VideoLayout
+        item={selectedItem}
+        title={inlineLayout ? selectedItem.title : seriesPlaylist.title}
+        description={selectedItem.description}
+        inlineLayout={inlineLayout}
+        primaryMetadata={primaryMetadata}
+        secondaryMetadata={secondaryMetadata}
+        image={selectedItem.backgroundImage}
+        shareButton={shareButton}
+        favoriteButton={favoriteButton}
+        trailerButton={trailerButton}
+        startWatchingButton={startWatchingButton}
+        isLoading={isLoading}
+        accessModel={accessModel}
+        isLoggedIn={isLoggedIn}
+        hasSubscription={hasSubscription}
+        playlist={playlist}
+        relatedTitle={inlineLayout ? seriesPlaylist.title : t('episodes')}
+        onItemClick={onCardClick}
+        setFilter={setSeasonFilter}
+        currentFilter={seasonFilter}
+        filterValuePrefix={t('season_prefix')}
+        defaultFilterLabel={t('all_seasons')}
+        activeLabel={t('current_episode')}
+        watchHistory={watchHistoryDictionary}
+        filterMetadata={filterMetadata}
+        filters={filters}
+        hasLoadMore={hasNextEpisodesPage}
+        loadMore={fetchNextEpisodes}
+        player={
+          inlineLayout && (episode || firstEpisode) ? (
+            <InlinePlayer
+              isLoggedIn={isLoggedIn}
+              item={episode || firstEpisode}
+              onComplete={handleComplete}
+              feedId={feedId ?? undefined}
+              startWatchingButton={startWatchingButton}
+              paywall={isLocked(accessModel, isLoggedIn, hasSubscription, episode || firstEpisode)}
+              autostart={play || undefined}
+            />
+          ) : (
+            <Cinema
+              open={play && isEntitled}
+              onClose={goBack}
+              item={episode || firstEpisode}
+              title={seriesPlaylist.title}
+              primaryMetadata={primaryMetadata}
+              secondaryMetadata={secondaryMetadata}
+              onComplete={handleComplete}
+              feedId={feedId ?? undefined}
+            />
+          )
+        }
+      />
+      {episode && <TrailerModal item={trailerItem} title={`${episode.title} - Trailer`} open={playTrailer} onClose={() => setPlayTrailer(false)} />}
+    </React.Fragment>
+  );
+};
 
 /**
- * This media screen is used to redirect a series linking media item to an episode page.
+ * This wrapper is used to check which series flow is used
+ * If only new flow with series api is used, it can be deleted and replaced by MediaSeriesContent
  */
-const MediaSeries: ScreenComponent<PlaylistItem> = ({ data, isLoading }) => {
-  const seriesId = getSeriesId(data) || '';
+const MediaSeries: ScreenComponent<PlaylistItem> = ({ data: media, isLoading: isMediaLoading }) => {
```

Comment:
I think if you make the change to isolate all the legacy stuff in the legacy screen then you can just put the rest of this logic at the top of the current component. Having 2 components like this is a little hard to follow.

Reply:
On the one hand it makes sense. However on the other hand having legacy logic in MediaSeriesContent is also something we should probably avoid.

Reply:
Totally agree. I would put it into the Legacy page (so initially it will redirect there w/o a series ID, but then the series ID will get populated)